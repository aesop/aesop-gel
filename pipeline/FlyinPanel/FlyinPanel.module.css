@import 'styles/variables/index';
@import 'styles/mixins/index';

.content {
  position: fixed;
  z-index: var(--z-index-flyin-panel);
  top: 0;
  bottom: 0;
  left: 0;
  justify-content: center;
  padding: 166px 40px 60px;
  background-color: var(--color-white-dark);
  overflow-y: auto;

  @media (--viewport-sm-only) {
    width: 100%;
  }

  @media (--viewport-md) {
    width: 50%;
  }
}

.closeButton {
  position: absolute;
  top: 18px;
  right: 20px;
}
