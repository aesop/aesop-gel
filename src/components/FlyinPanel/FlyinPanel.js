import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import uuidv4 from 'uuid/v4';
import { useEscapeKeyListener } from '~/hooks/useEscapeKeyListener';
import { useOverflowHidden } from '~/hooks/useOverflowHidden';
import Button from '~/components/Button';
import Heading from '~/components/Heading';
import Icon from '~/components/Icon';
import Overlay from '~/components/Overlay';
import Transition from '~/components/Transition';
import styles from './FlyinPanel.module.css';

const FlyinPanel = ({ children, className, heading, isVisible, onClose }) => {
  useEscapeKeyListener(onClose);
  useOverflowHidden(isVisible);

  const classSet = cx(styles.base, className);
  const labelledby = uuidv4();
  const describedby = uuidv4();
  const asideRole = 'dialog';
  const closeButtonTitle = 'Close';

  return (
    <>
      <Overlay isVisible={isVisible} onClose={onClose} />
      <Transition
        hasCSSTransitionMountOnEnter={true}
        hasCSSTransitionUnmountOnExit={true}
        isActive={!!isVisible}
        type="slideRight"
      >
        <aside
          aria-describedby={describedby}
          aria-hidden={!isVisible}
          aria-labelledby={labelledby}
          className={classSet}
          role={asideRole}
        >
          <Button
            className={styles.closeButton}
            isInline={true}
            onClick={onClose}
            tabIndex={0}
            title={closeButtonTitle}
          >
            <Icon height={12} name="close" width={12} />
          </Button>
          {heading && (
            <Heading id={labelledby} level="2" size="small">
              {heading}
            </Heading>
          )}
          <div id={describedby}>{children}</div>
        </aside>
      </Transition>
    </>
  );
};

FlyinPanel.propTypes = {
  children: PropTypes.any.isRequired,
  className: PropTypes.string,
  heading: PropTypes.string,
  isVisible: PropTypes.bool,
  onClose: PropTypes.func.isRequired,
};

FlyinPanel.defaultProps = {
  children: undefined,
  className: undefined,
  heading: undefined,
  isVisible: false,
  onClose: undefined,
};

export default FlyinPanel;
