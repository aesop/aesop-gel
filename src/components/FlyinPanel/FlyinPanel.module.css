@import 'styles/variables/index';
@import 'styles/mixins/index';

.base {
  position: fixed;
  z-index: var(--z-index-flyin-panel);
  top: 0;
  bottom: 0;
  left: 0;
  width: 100%;
  justify-content: center;
  padding-top: calc(var(--layout-sm-spacing) * 4);
  padding-right: var(--layout-sm-spacing);
  padding-bottom: calc(var(--layout-sm-spacing) * 2);
  padding-left: var(--layout-sm-spacing);
  background-color: var(--color-white-dark);
  font-family: var(--font-family-sans-serif-regular);
  font-size: rem(15px);
  font-weight: var(--font-weight-normal);
  line-height: 1.6;
  overflow-y: auto;

  &,
  & * {
    box-sizing: border-box;
  }

  @media (--viewport-md) {
    padding-top: calc(var(--layout-md-spacing) * 4);
    padding-right: calc(var(--layout-md-spacing) * 4);
    padding-bottom: calc(var(--layout-md-spacing) * 2);
    padding-left: calc(var(--layout-md-spacing) * 4);
  }

  @media (--viewport-lg) {
    width: 50%;
    padding-top: calc(var(--layout-lg-spacing) * 4);
    padding-right: calc(var(--layout-lg-spacing) * 2);
    padding-bottom: calc(var(--layout-lg-spacing) * 2);
    padding-left: calc(var(--layout-lg-spacing) * 2);
  }
}

.closeButton.closeButton {
  position: absolute;
  top: 18px;
  right: 20px;
  padding: 2px 4px 0 4px;
}
