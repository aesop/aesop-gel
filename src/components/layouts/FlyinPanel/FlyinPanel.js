import React from 'react';
import cx from 'classnames';
import uuidv4 from 'uuid/v4';
import { useEscapeKeyListener } from '~/hooks/useEscapeKeyListener';
import { useOverflowHidden } from '~/hooks/useOverflowHidden';
import Button from '~/components/elements/Button';
import Heading from '~/components/elements/Heading';
import Icon from '~/components/elements/Icon';
import Overlay from '~/components/elements/Overlay';
import Transition from '~/components/elements/Transition';
import PROP_TYPES from './FlyinPanel.prop-types';
import styles from './FlyinPanel.module.css';

const FlyinPanel = ({
  children,
  className,
  handleClose,
  heading,
  isVisible = false,
}) => {
  useEscapeKeyListener(handleClose);
  useOverflowHidden(isVisible);

  const classSet = cx(className);
  const labelledby = uuidv4();
  const describedby = uuidv4();
  const asideRole = 'dialog';
  const closeButtonTitle = 'Close';

  return (
    <aside
      aria-describedby={describedby}
      aria-hidden={!isVisible}
      aria-labelledby={labelledby}
      className={classSet}
      role={asideRole}
    >
      <Overlay handleClose={handleClose} isVisible={isVisible} />
      <Transition
        active={!!isVisible}
        mountOnEnter={true}
        type="slideRight"
        unmountOnExit={true}
      >
        <div className={styles.content}>
          <Button
            className={styles.closeButton}
            inline={true}
            onClick={handleClose}
            tabIndex={0}
            title={closeButtonTitle}
          >
            <Icon height={12} name="close" width={12} />
          </Button>
          {heading && (
            <Heading id={labelledby} level="2" size="small">
              {heading}
            </Heading>
          )}
          <div id={describedby}>{children}</div>
        </div>
      </Transition>
    </aside>
  );
};

FlyinPanel.propTypes = PROP_TYPES;

export default FlyinPanel;
