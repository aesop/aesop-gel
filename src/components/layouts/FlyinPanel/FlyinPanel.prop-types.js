import PropTypes from 'prop-types';

export const PROP_TYPES = {
  children: PropTypes.any.isRequired,
  className: PropTypes.string,
  handleClose: PropTypes.string.isRequired,
  heading: PropTypes.string,
  isVisible: PropTypes.boolean,
};

export default PROP_TYPES;
