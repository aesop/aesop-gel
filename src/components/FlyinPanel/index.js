import FlyinPanel from './FlyinPanel';

export default FlyinPanel;
