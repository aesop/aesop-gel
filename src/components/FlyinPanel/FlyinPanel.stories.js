import React from 'react';
import { storiesOf } from '@storybook/react';
import '~.storybook/storybook.module.css';
import { State } from '~.storybook/State';
import Button from '~/components/Button';
import Heading from '~/components/Heading';
import { P } from '~/components/Paragraph';
import FlyinPanel from './FlyinPanel';

storiesOf('Components/FlyinPanel', module).add('Base component', () => (
  <State initialState={{ isVisible: false }}>
    {(state, setState) => (
      <>
        <FlyinPanel
          isVisible={state.isVisible}
          onClose={() => {
            setState(() => ({ ...state, isVisible: false }));
          }}
          title="Ingredients"
        >
          <Heading level="2" size="xSmall">
            Product Ingredients
          </Heading>
          <P>
            Water (Aqua), Glycerin, Sodium Ascorbyl Phosphate, Niacinamide,
            PEG-40 Hydrogenated Castor Oil, Phenoxyethanol, Citric Acid, Xanthan
            Gum, Sodium Gluconate, Sodium Dehydroacetate, Bisabolol, Polysorbate
            20, Boswellia Carterii Oil, Fusanus Spicatus Wood Oil, Rosa
            Damascena Flower Oil, Biosaccharide Gum-1, Sodium Carrageenan,
            Glyceryl Caprylate, Sodium Anisate, Sodium Levulinate, Sea Salt
            (Maris sal), d-Limonene, Citronellol, Farnesol, Geraniol.
          </P>
          <P>
            This ingredient list is subject to change, customers should refer to
            the product packaging for the most up-to-date ingredient list.
          </P>
        </FlyinPanel>
        <Button
          onClick={() => {
            setState(() => ({ ...state, isVisible: true }));
          }}
          title="open Flyin"
        >
          Open Flyin
        </Button>
      </>
    )}
  </State>
));
