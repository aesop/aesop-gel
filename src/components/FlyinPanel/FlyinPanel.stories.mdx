import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import { withKnobs } from '@storybook/addon-knobs';
import * as knobs from '@storybook/addon-knobs';
import { State } from '~.storybook/State';
import Button from '~/components/Button';
import Heading from '~/components/Heading';
import { P } from '~/components/Paragraph';
import FlyinPanel from './FlyinPanel';
import FlyinPanelFixture from './FlyinPanel.fixture';

<Meta
  title="Components/FlyinPanel"
  component={FlyinPanel}
  decorators={[withKnobs]}
/>

# FlyinPanel

<Preview>
  <Story name="Base component">
    <State initialState={{ isVisible: false }}>
      {(state, setState) => {
        return (
          <React.Fragment>
            <FlyinPanel
              isVisible={state.isVisible}
              onClose={() => {
                setState(() => ({ ...state, isVisible: false }));
              }}
              title="Ingredients"
            >
              <Heading level="2" size="xSmall">
                Product Ingredients
              </Heading>
              <P>
                Water (Aqua), Glycerin, Sodium Ascorbyl Phosphate, Niacinamide,
                PEG-40 Hydrogenated Castor Oil, Phenoxyethanol, Citric Acid, Xanthan
                Gum, Sodium Gluconate, Sodium Dehydroacetate, Bisabolol, Polysorbate
                20, Boswellia Carterii Oil, Fusanus Spicatus Wood Oil, Rosa
                Damascena Flower Oil, Biosaccharide Gum-1, Sodium Carrageenan,
                Glyceryl Caprylate, Sodium Anisate, Sodium Levulinate, Sea Salt
                (Maris sal), d-Limonene, Citronellol, Farnesol, Geraniol.
              </P>
              <P>
                This ingredient list is subject to change, customers should refer to
                the product packaging for the most up-to-date ingredient list.
              </P>
            </FlyinPanel>
            <Button
              onClick={() => {
                setState(() => ({ ...state, isVisible: true }));
              }}
              title="open Flyin"
            >
              Open Flyin
            </Button>
          </React.Fragment>
        );
      }}
    </State>
  </Story>
</Preview>

## API

<Props of={FlyinPanel} />
