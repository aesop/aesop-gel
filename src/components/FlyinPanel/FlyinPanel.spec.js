import React from 'react';
import { render } from '@testing-library/react';
import FlyinPanel from './FlyinPanel';

jest.mock('uuid', () => {
  let value = 0;
  return {
    v4: () => value++,
  };
});

const mockFn = jest.fn();

describe('<FlyinPanel />', () => {
  it('should be defined', () => {
    expect(FlyinPanel).toBeDefined();
  });

  it('renders base component correctly', () => {
    const { container } = render(
      <FlyinPanel onClose={mockFn} title="title">
        Content
      </FlyinPanel>,
    );

    expect(container).toMatchSnapshot();
  });
});
