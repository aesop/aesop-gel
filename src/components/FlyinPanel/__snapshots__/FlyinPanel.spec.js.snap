// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlyinPanel /> renders base component correctly 1`] = `<div />`;
